### PR TITLE
Implementation for concat relu operator

### PR DESCRIPTION
**Pull request by EvanzzzZ**
Implement concat relu operator as described in the [paper](https://arxiv.org/pdf/1603.05201.pdf). 
The paper claim fewer parameters with better accuracies by replacing relu unit with concat relu unit in CNN.

Unit test passed and run successfully with cpu context.

@piiswrong

### CODE REVIEW

File: src/operator/contrib/crelu-inl.h [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                       const std::vector<TBlob> &aux_args) {
+    using namespace mshadow;
+    using namespace mshadow::expr;
+    CHECK_EQ(in_data.size(), 1U);
+    CHECK_EQ(out_data.size(), 1U);
+    Stream<xpu> *s = ctx.get_stream<xpu>();
+
+    Tensor<xpu, 2, DType> data = in_data[concat_relu::kData].FlatTo2D<xpu, DType>(s);
+    Tensor<xpu, 2, DType> out = out_data[concat_relu::kOut].FlatTo2D<xpu, DType>(s);
+    std::vector<Tensor<xpu, 2, DType> > concat_data(2);
```
Comment (piiswrong) [inline src/operator/contrib/crelu-inl.h]: Mshadow doesn't work like this. You can't create tensors and assign to them.

Comment (EvanzzzZ) [inline src/operator/contrib/crelu-inl.h]: I saw similar code in `concat-inl.h`, which create a vector of tensors and then assign each of the tensor.
L55 just specify the type of elements, L57 and L58 specify the pointer/memory space for the tensors.

**Comment by chinakook** [comment]
@EvanzzzZ ,Awesome feature!!!
With CReLU, the Face detection make SOTA in real time on CPU. https://arxiv.org/pdf/1708.05234.pdf

**Comment by piiswrong** [comment]
closing due to inactive

**Pull request closed without merging**